jax/_src/lax/control_flow.py

```diff
@@ -1936,6 +1936,30 @@ def body(i, dst):
 masking.masking_rules[lax.concatenate_p] = _concat_masking_rule  # type: ignore
 
 
+def _pad_batch_rule(batched_args, batch_dims, *, padding_config):
+  operand, padding_value = batched_args
+  operand_bdim, padding_value_bdim = batch_dims
+  padding_config = list(padding_config)
+  if padding_value_bdim is None:
+    assert operand_bdim is not None
+    padding_config.insert(operand_bdim, (0, 0, 0))
+    return lax.pad(operand, padding_value, padding_config), operand_bdim
+  else:
+    # Different padding values for each batch element. Loop and stack with scan
```

Comment:
Is there a better way of padding different rows with different values, without a loop?

Reply:
I think there is. In the absence of interior padding, you can broadcast the padding values to a suitable shape (i.e., with 1s in all the other dimensions of the LHS), replicate them as many times as the pad config says,  and concatenate them on the left and right.

To handle interior padding, I think you could pad an array of `True`s padded with `False` values to get a mask saying which elements are padding. You can then select either the appropriate padding value or data value (which you would also need to dilate using `pad` so it lines up, with any padding value) depending on the mask.

Reply:
https://github.com/google/jax/pull/5762 implements an idea close to the second paragraph here. I think it's much simpler so we can close this PR.